loader/src/utils/web.cpp

```diff
@@ -212,7 +212,9 @@ class WebRequest::Impl {
     }
 };
 
-WebRequest::WebRequest() : m_impl(std::make_shared<Impl>()) {}
+size_t WebRequest::m_idCounter = 0;
+
+WebRequest::WebRequest() : m_impl(std::make_shared<Impl>()), m_id(WebRequest::m_idCounter++) {}

```

Comment:
This is a race condition, WebRequests aren't necessarily created in the same thread